search/indexing_api.py

```diff
@@ -44,25 +43,35 @@
     'created': {'type': 'date'},
     'deleted': {'type': 'boolean'},
     'removed': {'type': 'boolean'},
-    'parent_post_removed': {'type': 'boolean'},
     'post_id': {'type': 'keyword'},
     'post_title': {'type': 'text'},
-    'post_link_url': {'type': 'keyword'},
-    'post_link_thumbnail': {'type': 'keyword'},
-    'num_comments': {'type': 'long'},
-    'comment_id': {'type': 'keyword'},
-    'parent_comment_id': {'type': 'keyword'},
+}
+
+MAPPING = {
+    POST_TYPE: dict(BASE_CONTENT_TYPE, **{
+        'post_link_url': {'type': 'keyword'},
+        'post_link_thumbnail': {'type': 'keyword'},
+        'num_comments': {'type': 'long'},
+    }),
+    COMMENT_TYPE: dict(BASE_CONTENT_TYPE, **{
+        'comment_id': {'type': 'keyword'},
+        'parent_comment_id': {'type': 'keyword'},
+        'parent_post_removed': {'type': 'boolean'},
+    }),
 }
 
 
-def clear_and_create_index(*, index_name=None, skip_mapping=False):
+def clear_and_create_index(*, index_name=None, skip_mapping=False, object_type=None):
     """
     Wipe and recreate index and mapping. No indexing is done.
 
     Args:
         index_name (str): The name of the index to clear
         skip_mapping (bool): If true, don't set any mapping
+        object_type(str): The type of document (post, comment)
     """
+    if not object_type or object_type not in VALID_OBJECT_TYPES:
+        raise TypeError('A valid object type must be specified when clearing and creating an index')
```

Comment:
This is pretty nitpicky but we should use `ValueError` here. According to [the docs](https://docs.python.org/3/library/exceptions.html#TypeError) `TypeError` is only if the Python types don't match with each other

Reply:
✅